dds_cli/__main__.py

```diff
@@ -1033,134 +828,323 @@ def abort(click_ctx, username, project):
+                            )
+
+                            # Get result
+                            try:
+                                file_downloaded = dfut.result()
+                                LOG.debug(
+                                    f"Download of {downloaded_file} successful: {file_downloaded}"
+                                )
+                            except concurrent.futures.BrokenExecutor as err:
+                                LOG.critical(
+                                    f"Download of file {downloaded_file} failed! Error: {err}"
+                                )
+                                continue
+
+                            new_tasks += 1
+                            progress.advance(task_dwnld)
+
+                        # Schedule the next set of futures for download
+                        for next_file in itertools.islice(iterator, new_tasks):
+                            LOG.debug(f"Starting: {next_file}")
+                            # Execute download
+                            download_threads[
+                                texec.submit(
+                                    getter.download_and_verify,
+                                    file=next_file,
+                                    progress=progress,
+                                )
+                            ] = next_file
     except (
+        dds_cli.exceptions.InvalidMethodError,
+        OSError,
+        dds_cli.exceptions.TokenNotFoundError,
         dds_cli.exceptions.AuthenticationError,
-        dds_cli.exceptions.ApiResponseError,
         dds_cli.exceptions.ApiRequestError,
+        dds_cli.exceptions.ApiResponseError,
+        SystemExit,
         dds_cli.exceptions.DDSCLIException,
+        dds_cli.exceptions.NoDataError,
+        dds_cli.exceptions.DownloadError,
     ) as err:
         LOG.error(err)
         sys.exit(1)
 
 
-# SUBCOMMAND#######################################################################
-####### DELETE_USERS ############################################### DELETE_USERS #
-###################################################################################
-## Delete users from the system
-@user.command(name="delete", no_args_is_help=True)
-@click.argument("email", nargs=1, type=str, required=False)
+# -- dds data ls -- #
+@data_group_command.command(name="ls", no_args_is_help=True)
+# Options
+@username_option()
+@project_option(required=True)
+@folder_option(help_message="List contents in this project folder.")
+# Flags
+@json_flag(help_message="Output in JSON format")
+@size_flag(help_message="Show size of project contents.")
+@tree_flag(help_message="Display the entire project(s) directory tree")
+@users_flag(help_message="Display users associated with a project(Requires a project id)")
+@click.pass_context
+def list_data(ctx, username, project, folder, json, size, tree, users):
+    """List project contents. Same as dds ls [PROJECT ID]."""
```

Comment:
I don't think dds ls accepts the project ID as an argument? At least the decorator is missing if you intend to have this functionality.

Reply:
I'm not sure I follow, just gonna reply a bit here, hope this clears it up:

This is `dds data ls` which takes the project id as an option. So the call should be `dds data ls --project`. This will list the project contents by calling the `dds ls` command which takes the project option too. Before it was an argument but in the `dds-development` channel we said that we should have the project ID as _either_ an positional argument or an option, and since it doesn't really make sense to have it as an argument in most cases, or at least just adds a bit to the confusion, I decided to add the `project_option` to `dds ls` instead.

Reply:
I was just referring to the help text that I found to be in disagreement with the way the command worked and wanted to make sure that the command is not intended to accept an argument. 

Yesterday I went though the various subcommands and tried to spot if I noticed any missing options or help texts that didn't fit the functionality. Here, I would just have expected _Same as dds ls -p [PROJECT ID]_ - so nothing serious!